src/jarabe/model/bundleregistry.py

```diff
@@ -140,6 +140,23 @@ def __file_monitor_changed_cb(self, monitor, one_file, other_file,
             self.add_bundle(one_file.get_path(), set_favorite=True)
         elif event_type == Gio.FileMonitorEvent.DELETED:
             self.remove_bundle(one_file.get_path())
+            for root in GLib.get_system_data_dirs():
+                root = os.path.join(root, 'sugar', 'activities')
+
+                try:
+                    dir_list = os.listdir(root)
+                except OSError:
+                    logging.debug('Can not find GLib system dir %s', root)
+                    continue
+                activity_dir = os.path.basename(one_file.get_path())
+                try:
+                    bundle = bundle_from_dir(os.path.join(root, activity_dir))
+                except MalformedBundleException:
+                    continue
+
+                path = bundle.get_path()
+                if path is not None:
+                    self.add_bundle(path)
```

Comment:
The bundleregistry `uninstall` method does the logic a little differently:

```
        alt_bundles = self.get_system_bundles(act.get_bundle_id())              
        if alt_bundles:                                                         
            alt_bundles.sort(                                                   
                key=lambda b: NormalizedVersion(b.get_activity_version()))      
            alt_bundles.reverse()                                               
            new_bundle = alt_bundles[0]                                         
            self.add_bundle(new_bundle.get_path())                              
```

Can we unify the logic in a shared function?

Other than that, thanks for the patches!  I love the sugar-erase-bundle script.